### PR TITLE
Enable StorageClass as option

### PR DESCRIPTION
**Pull request by scheeles**
Now it possible to provide an own StorageClass with more detail options as an alternative to the PV-Provisioner

### CODE REVIEW

**Comment by hongchaodeng** [comment]
@scheeles 

What is the purpose of adding storage class option? Any use case?

**Comment by scheeles** [comment]
We want to have more control over the storage. On AWS e.g. we want to use `encrypted` and control the `type`. Additional we use the etcdOperator on Glusterfs and have backup in place.

**Comment by hongchaodeng** [comment]
@scheeles Can you create an issue to discuss your use case first?

I think this feature needs some additional thoughts:
- It's duplicate to have both "pv-provisioner" and "storageclass" options.
- See https://github.com/coreos/etcd-operator/issues/691 . Even if we provide storage class, is it gonna be per cluster level? This is more flexible.
- See https://github.com/coreos/etcd-operator/issues/691#issuecomment-288184323 . There is a hack on naming "etcd-backup-gce-pd" to your custom storage class currently.

File: cmd/operator/main.go [hunk 1]
```diff
@@ -68,6 +69,7 @@ func init() {
 	flag.BoolVar(&analyticsEnabled, "analytics", true, "Send analytical event (Cluster Created/Deleted etc.) to Google Analytics")
 
 	flag.StringVar(&pvProvisioner, "pv-provisioner", constants.PVProvisionerGCEPD, "persistent volume provisioner type")
+	flag.StringVar(&storageClass, "storageclass", "", "Storage class type")
```
Comment (xiang90) [inline cmd/operator/main.go]: operator probably should die with error message when both storage class and pv-provisioner are provided.

also this flag needs a better help message. storage class type is too generic.

Comment (xiang90) [inline cmd/operator/main.go]: the pv-provisioner flag message is also bad :(

File: pkg/cluster/backupstorage/pv.go [hunk 2]
```diff
@@ -11,23 +11,25 @@ type pv struct {
 	clusterName   string
 	namespace     string
 	pvProvisioner string
+	storageClass  string
```
Comment (xiang90) [inline pkg/cluster/backupstorage/pv.go]: you can remove the pvProvisioner string actually. i think we actually convert the provisioner to storageclass anyway

**Comment by jamiehannaford** [comment]
@scheeles Are you still working on this? If not, I can cherry pick your work and fix the final details.

I'd like to add Cinder as a new PV option, so this PR would supersede my one (https://github.com/coreos/etcd-operator/pull/1351).

/cc @hongchaodeng

**Comment by hongchaodeng** [comment]
@jamiehannaford 
This issue has been inactive for a while. Let's just close it and let you take over it.
Note that this is still changing global flag, not adding option in cluster spec.
Thanks for taking care of it!

**Pull request closed without merging**